### PR TITLE
feat: message sharing attachments permissions request (AR-1098)

### PR DESCRIPTION
**Pull request by yamilmedina**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-1098" title="AR-1098" target="_blank">SubtaskAR-1098</a>  Implement runtime permissions flow for file reading & writing
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [X] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [X] contains a reference JIRA issue number like `SQPIT-764`
  - [X] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [X] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

https://wearezeta.atlassian.net/browse/AR-1098

Permissions for all the options displayed as possible attachments, this includes:
- Storage access
- Camera access
- Location access
- Mic access

### Attachments (Optional)

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [X] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [X] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: app/src/main/kotlin/com/wire/android/util/permission/RecordAudioRequestFlow.kt [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    fun launch() {
+        if (context.checkPermission(android.Manifest.permission.RECORD_AUDIO) &&
+            context.checkPermission(android.Manifest.permission.WRITE_EXTERNAL_STORAGE)
+        ) {
+            // TODO: launch record audio flow
+        } else {
+            audioRecordPermissionLauncher.launch(
+                arrayOf(
+                    android.Manifest.permission.RECORD_AUDIO,
+                    android.Manifest.permission.WRITE_EXTERNAL_STORAGE
```
Comment (gongracr) [inline app/src/main/kotlin/com/wire/android/util/permission/RecordAudioRequestFlow.kt:49]: Would we really need to save the audio recordings on external storage? or should we proceed as with the image sharing, storing them temporarily in internal storage no matter whether they are picked from gallery or taken by the camera?

**Review by gongracr — COMMENTED** [review]
Looking great, just a small out loud question about how to proceed with audio recording storage before sharing 🤔

File: app/src/main/kotlin/com/wire/android/util/permission/RecordAudioRequestFlow.kt (hunk 1 above)
Comment (yamilmedina) [reply app/src/main/kotlin/com/wire/android/util/permission/RecordAudioRequestFlow.kt]: I think probably we'll take the same approach when implementing the specific sharing features.  We'll have to create a temp file for every kind of attachment and just overwrite that when we want to send the N+1 audio file, for example.

Changed to `READ_EXTERNAL_STORAGE` ⚡

**Review by gongracr — APPROVED** [review]
Great job on reusing all these picker and request flows! 🚆🛣️

**Pull request merged**